x-pack/plugins/rule_registry/server/alert_data_client/alerts_client.ts

```diff
@@ -616,7 +616,7 @@ export class AlertsClient {
 
       const toReturn = Array.from(authorizedFeatures).flatMap((feature) => {
         if (isValidFeatureId(feature)) {
-          return mapConsumerToIndexName[feature];
+          return `${mapConsumerToIndexName[feature]}-${this.spaceId}`;
```

Comment:
I am worried that will not always be the case for o11y, I agree it work for security solution

Reply:
👍🏽 updated a4e10f5